### PR TITLE
misc. coord cleanup

### PR DESCRIPTION
**Pull request by sploiselle**
This isn't very meaningful on its own, but is an attempt to atomize #5445.

These commits:
- Clean up some error handling that was growing unwieldy
- Defer printing `coord::catalog::CatalogItem` to `sql::catalog::CatalogItemType`

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5504)

### CODE REVIEW

File: src/sql/src/catalog.rs [hunk 1]
```diff
@@ -262,20 +263,16 @@ impl fmt::Display for CatalogError {
...
-                f,
-                "catalog item '{}' is a type and so cannot be depended upon",
-                name
+                "catalog item '{}' is {} {} and so cannot be depended upon",
+                name,
+                if matches!(typ, CatalogItemType::Index) {
+                    "an"
+                } else {
+                    "a"
+                },
```
Comment (benesch) [inline src/sql/src/catalog.rs:274]: Grammar is the worst!

File: src/coord/src/catalog.rs [hunk 2]
```diff
@@ -28,7 +28,9 @@ use expr::{ExprHumanizer, GlobalId, MirScalarExpr, OptimizedMirRelationExpr};
 use repr::{ColumnType, RelationDesc, ScalarType};
 use sql::ast::display::AstDisplay;
 use sql::ast::{Expr, Raw};
-use sql::catalog::{Catalog as SqlCatalog, CatalogError as SqlCatalogError};
+use sql::catalog::{
+    Catalog as SqlCatalog, CatalogError as SqlCatalogError, CatalogItem as SqlCatalogItem,
```
Comment (benesch) [inline src/coord/src/catalog.rs:32]: So so minor, but you went to all this trouble, you might as well import `CatalogItemType as SqlCatalogItemType`!

**Review by benesch — APPROVED** [review]
Looks great to me!

File: src/sql/src/catalog.rs (hunk 1 above)
Comment (sploiselle) [reply src/sql/src/catalog.rs:274]: Or is the problem that computers are too simple to keep up? 🤔

File: src/coord/src/catalog.rs (hunk 2 above)
Comment (sploiselle) [reply src/coord/src/catalog.rs:32]: Will avoid a trip through CI and put this in my next PR.

**Pull request merged**